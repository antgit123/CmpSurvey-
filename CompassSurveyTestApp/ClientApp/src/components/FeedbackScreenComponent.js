import React, { Component } from 'react';
import { Button,Row } from 'react-bootstrap';
import { Link } from 'react-router-dom';

export class FeedbackScreenComponent extends Component {

    constructor(props) {
        super(props);
    }

    render() {
        const { loading, message, backButtonUrl } = this.props;
        if (loading) {
            return (
                <div>
                    {message}
                </div>
            )                           
        }
        return (
            <div className="feedbackDiv">
                <Row>
                    <p>{message}</p>
                </Row>
                <Row>
                    <Link to={backButtonUrl} >
                        <Button variant="primary">Back</Button>
                    </Link>
                </Row>                
             </div>
        )        
    }
}