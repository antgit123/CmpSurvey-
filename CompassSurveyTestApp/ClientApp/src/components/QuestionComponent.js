import React, { Component } from 'react';

export default class QuestionComponent extends Component {

    constructor(props) {
        super(props);
    }

    render() {
        const { question } = this.props;
        //input type can be anything and can be changed to support generic form with different inputs
        const inputType = "radio";
        return (
            <div className="form-group questionBox">
                <label htmlFor={question.id} > Question {question.id}</label>
                <p> {question.title} </p>
                {
                        question.options.map((option) => (                          
                        <div className="form-check" key={option.id.toString()}>
                            <input className="form-check-input radioStyle" type={inputType} name={"question" + question.id + option.id + "flexRadio"} id={"question" + question.id + option.id} />
                            <label className="form-check-label" htmlFor={"question" + question.id + option.id}>
                                        {option.optionText}
                                    </label>
                            </div>
                        ))
                }                   
            </div>
        )
    }

}
