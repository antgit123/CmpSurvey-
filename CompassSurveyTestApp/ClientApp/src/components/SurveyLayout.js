import React, { Component } from 'react';
import { Col, Grid, Row } from 'react-bootstrap';
import { NavMenu } from './NavMenu';

export class SurveyLayout extends Component {
    displayName = SurveyLayout.name

    render() {
        return (
            <Grid fluid>
                <Row>
                    <Col sm={2}>
                        
                    </Col>                    
                    <Col sm={8}>
                        <div className="appContainer"> 
                            {this.props.children}
                        </div>
                    </Col>
                    <Col sm={2}/>
                </Row>
            </Grid>
        );
    }
}
