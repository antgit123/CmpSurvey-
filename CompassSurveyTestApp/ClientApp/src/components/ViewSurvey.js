import React, { Component } from 'react';
import { Button } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import { FeedbackScreenComponent } from './FeedbackScreenComponent';

export class ViewSurvey extends Component {
    
    loadSurveyData = () => {         
        fetch('api/Surveys')
            .then(response => response.json())
            .then(data => {
                this.setState({ surveyData: data, loading: false });
            });  
    }

    onSelectSurvey = (survey) => {
        this.setState({ selectedSurvey: survey });
    }
    constructor(props) {
        super(props);
        this.state = { surveyData: [], selectedSurvey: null, loading: true };   
    }

    componentDidMount() {
        this.loadSurveyData();          
    }
    render() {
        const { loading, surveyData } = this.state;
        //Show feedback screen to the user if the fetching of surveys hasn't completed
        if (loading) {           
           return( 
               <FeedbackScreenComponent
                    message={`Survey Data loading...`}
                    loading={true}                    
                />
            )
        }
        // check data after it is returned, If no surveys found display feedback to the user with the option to navigate back
        if (!loading && (!surveyData.surveys || surveyData.surveys.length === 0)) {            
            return (
                <FeedbackScreenComponent
                    message={`Sorry, there is no list of surveys. Please check back later`}                    
                    loading={false}
                    backButtonUrl={`/`}
                />
            )
        }
        //Display list of surveys to the user if the above prerequisites are satisfied 
        return (
            <div>
                <h3 className="appHeader"> Compass Surveys </h3>
                {surveyData.surveys.map((survey) => (
                    <Link to={{
                        pathname: `/surveys/${survey.id}`,
                        state: {
                            selectedSurvey: survey
                        }
                    }}>
                        <Button className="surveyButton" key={survey.id} variant="primary" size="lg" block onClick={this.onSelectSurvey.bind(this, survey)}>{survey.name}</Button>
                    </Link>
                ))}
            </div>    
        );
    }
}