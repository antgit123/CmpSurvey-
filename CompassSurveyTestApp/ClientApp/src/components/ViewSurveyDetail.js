import React, { Component } from 'react';
import QuestionComponent from './QuestionComponent';
import { FeedbackScreenComponent } from './FeedbackScreenComponent';
import { Form, Button } from 'react-bootstrap';
import { Link } from 'react-router-dom';

 export class ViewSurveyDetail extends Component {

    constructor(props) {
        super(props);  
        this.state = { surveyQuestionData: [], loading: true };                        
     }

     componentDidMount() {
         const { selectedSurvey } = this.props.location.state;
         this.loadSurveyQuestionData(selectedSurvey);         
     }

     loadSurveyQuestionData(selectedSurvey) {
        // API call to fetch questions from the selected survey 
        fetch('api/Surveys/' + selectedSurvey.id)
            .then(response => response.json())
            .then(data => {
                this.setState({ surveyQuestionData: data, survey: selectedSurvey, loading: false });
            });  
     }

     render() {         
         //If survey questions are loading show this screen to the user to indicate that application is loading         
         if (this.state.loading) {             
             return (
                 <FeedbackScreenComponent
                     message={`Survey Questions Loading...`}
                     loading={true}
                 />
             )
         }
         // check data after it is returned, If no questions found display feedback to the user else show screen with all questions
         const surveyQuestions = this.state.surveyQuestionData;
         const hasNoQuestions = !surveyQuestions || surveyQuestions.length === 0;
         if (!this.state.loading && hasNoQuestions) {             
             return (
                 <FeedbackScreenComponent
                     message={`Sorry no questions found in the survey. Please choose another survey`}
                     loading={false}                     
                     backButtonUrl={`/surveys`}
                 />
             )
         }
         //load all questions from the survey if above preconditions are satisfied 
         return (
             <div>
                 <h3 className="appHeader">{this.state.survey.name}</h3>
                    <Form>
                    {
                            surveyQuestions.map((question) => (    
                                <QuestionComponent key={question.id.toString()} question={question}/>                                                                
                            ))}
                        <Link to="/surveys">
                            <Button variant="primary">Back</Button>
                        </Link>
                 </Form>
             </div>   
            )          
     }
}

