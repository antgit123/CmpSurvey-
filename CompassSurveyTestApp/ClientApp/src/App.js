import React, { Component } from 'react';
import { Route } from 'react-router';
import { Home } from './components/Home';
import { ViewSurvey } from './components/ViewSurvey';
import { ViewSurveyDetail } from './components/ViewSurveyDetail';
import { SurveyLayout } from './components/SurveyLayout';

export default class App extends Component {
  displayName = App.name

  render() {
    return (
      <SurveyLayout>            
            <Route exact path='/' component={Home} />      
            <Route exact path='/surveys' component={ViewSurvey} />
            <Route path='/surveys/:id' component={ViewSurveyDetail} />
        </SurveyLayout>
    );
  }
}
