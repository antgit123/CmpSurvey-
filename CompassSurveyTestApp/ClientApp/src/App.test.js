import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { Home } from './components/Home';
import { ViewSurvey } from './components/ViewSurvey';
import { data } from './NewFolder/staticData';
import { ViewSurveyDetail } from './components/ViewSurveyDetail';
import { QuestionComponent } from './components/QuestionComponent';
import { shallow,configure, mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
import { Button } from 'react-bootstrap';

configure({ adapter: new Adapter() });
//Test-1 : Check if the App component renders properly
it('renders without crashing', () => {
  const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
        <App />
    </BrowserRouter>,
        div);
});

//Test-2 : Check if home component rendered 
it('render Home component', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
});

//Test-2 : Check if home component has the survey list link 
it('render Home component', () => {
    const wrapper = shallow(<Home />);
    const surveyLink = '<li><strong>Survey List</strong></li>';
    expect(wrapper.contains(surveyLink)).toEqual(true);
});

it('Check if View Survey lists surveys when data loading has completed', () => {
    const props = {
        loading: false
    };
    const fetch = jest.fn(() => Promise.resolve());
    const mockSuccessResponse = data;
    const mockJsonPromise = Promise.resolve(mockSuccessResponse); 
    const mockFetchPromise = Promise.resolve({ 
        json: () => mockJsonPromise,
    });
    global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);
    const wrapper = shallow(<ViewSurvey />);
    const buttonComponent = wrapper.find(<button></button>);
    expect(buttonComponent.hasClass('surveyButton')).toEqual(true);
});


it("Expect components loaded", () => {
    const wrapper = shallow(<Home />);
    const header = <h1>Compass Surveys!</h1>;
    const home = wrapper.contains(<Home/>);
    //const surveyListLink = <li><strong>Survey List</strong></li>;
    expect(home).toEqual(true);
    //expect(wrapper.find(surveyListLink)).toEqual(true);
});

//Test-2 DOM test check if home page loaded with header and survey list 
it("displays the header and survey list on home screen", () => {
    const wrapper = shallow(<App />);
    const header = <h1>Compass Surveys!</h1>;
    const surveyListLink = <li><strong>Survey List</strong></li>;
    expect(wrapper.find(header)).toEqual(true);
    expect(wrapper.find(surveyListLink)).toEqual(true);
});
