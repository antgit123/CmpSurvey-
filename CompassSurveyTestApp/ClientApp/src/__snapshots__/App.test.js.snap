// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Home component 1`] = `ShallowWrapper {}`;
